src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,17 +3,56 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE_ARRAY = 10;
+    private K[] keys;
+    private V[] values;
+    private int size;
+
+    public StorageImpl() {
+        keys = (K[]) new Object[MAX_SIZE_ARRAY];
+        values = (V[]) new Object[MAX_SIZE_ARRAY];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        if (index != -1) {
+            values[index] = value;
+        } else {
+            if (size < MAX_SIZE_ARRAY) {
+                keys[size] = key;
+                values[size] = value;
+                size++;
+            } else {
+                throw new IllegalStateException("Storage is full");
+            }
```

Comment:
It's better to keep code with minimum tabulations to the right where possible:
```suggestion
            if (size >= MAX_SIZE_ARRAY) {
                throw new IllegalStateException("Storage is full");
            } 
            keys[size] = key;
            values[size] = value;
            size++;
            }
```